### PR TITLE
feat: Transform Reset method for different spaces 

### PR DESCRIPTION
**Pull request by Yurii-Tor**
## Purpose of this PR
It's often used for both world and local spaces, so both of them are implemented.

## Testing status
No Unit tests

### Manual testing status
Tested via Unity Editor

### CODE REVIEW

**Comment by Yurii-Tor** [comment]
I would like to discuss that Codacy Code Analysis

File: Runtime/Extensions/UnityEngine/TransformExtensions.cs [hunk 1]
```diff
@@ -23,11 +23,23 @@ public static void SetLossyScale(this Transform transform, Vector3 lossyScale)
         /// Reset <see cref="Transform"/> component position, scale and rotation.
         /// </summary>
         /// <param name="transform">Transform component.</param>
-        public static void Reset(this Transform transform)
```
Comment (stan-osipov) [inline Runtime/Extensions/UnityEngine/TransformExtensions.cs:26]: We can not remove it. This is a breaking change. 
Add obsolete attribute.

**Review by stan-osipov — CHANGES_REQUESTED** [review]
Meet Codacy request and good to go

File: Runtime/Extensions/UnityEngine/TransformExtensions.cs [hunk 2]
```diff
@@ -23,11 +23,25 @@ public static void SetLossyScale(this Transform transform, Vector3 lossyScale)
...
+                    transform.localScale = Vector3.one;
+                    transform.localPosition = Vector3.zero;
+                    transform.localRotation = Quaternion.identity;
+                    break;
+                case Space.World:
+                    transform.SetPositionAndRotation(Vector3.zero, Quaternion.identity);
+                    transform.SetLossyScale(Vector3.one);
+                    break;
+                default:
+                    throw new System.ArgumentException("Parameter 'relativeTo' contains unknown value. 'Space.Self' and 'Space.World' are accepted", nameof(relativeTo));
```
Comment (stan-osipov) [inline Runtime/Extensions/UnityEngine/TransformExtensions.cs:43]: Just a super small final comment. Would be nice to use `nameof(relativeTo)`, and the same for `Space.Self`, etc.

So what I mean is 
```$"Parameter '{nameof(relativeTo)}' contains...."```

**Review by stan-osipov — CHANGES_REQUESTED** [review]
thx for the changes! 
One small final request.

File: Runtime/Extensions/UnityEngine/TransformExtensions.cs (hunk 2 above)
Comment (Yurii-Tor) [reply Runtime/Extensions/UnityEngine/TransformExtensions.cs]: Done!

**Review by stan-osipov — APPROVED** [review]
looks good!

**Pull request merged**